flight/mcu/lib/teensyShared/ActiveControl.cpp

```diff
@@ -9,38 +9,22 @@
 #include "ActiveControl.h"
 
 #include "rocketFSM.h"
-#include "thresholds.h"
+#include "kalmanFilter.h"
 
-Controller::Controller(struct pointers* pointer_struct, PWMServo* controller_servo)
-    : activeControlServos(controller_servo) {
-    controller_servo_ = controller_servo;
-    stateData_ = &pointer_struct->sensorDataPointer->state_data;
-    current_state = &pointer_struct->sensorDataPointer->rocketState_data.rocketStates[0];
-    b_alt = &pointer_struct->sensorDataPointer->barometer_data.altitude;
-    dataMutex_barometer_ = &pointer_struct->dataloggerTHDVarsPointer.dataMutex_barometer;
-    dataMutex_state_ = &pointer_struct->dataloggerTHDVarsPointer.dataMutex_state;
+Controller activeController;
 
-    /*
-     * Startup sequence
-     * 15 degrees written to servo since this was
-     * experimentally determined to be the position in which
-     * the flaps are perfectly flush with the airframe.
-     */
-    controller_servo_->write(120);
-    chThdSleepMilliseconds(1000);
-    controller_servo_->write(12);
-    chThdSleepMilliseconds(1000);
+Controller::Controller() : activeControlServos(&controller_servo_) { }
 
-    setLaunchPadElevation();
-}
+void Controller::ctrlTickFunction() {
+    chMtxLock(&kalmanFilter.mutex);
+    array<float, 2> init = {kalmanFilter.getState().x, kalmanFilter.getState().vx};
+    chMtxUnlock(&kalmanFilter.mutex);
 
-void Controller::ctrlTickFunction(pointers* pointer_struct) {
-    chMtxLock(dataMutex_state_);
-    array<float, 2> init = {stateData_->state_x, stateData_->state_vx};
-    chMtxUnlock(dataMutex_state_);
     float apogee_est = rk4_.sim_apogee(init, 0.3)[0];
 
-    stateData_->state_apo = apogee_est;
+    chMtxLock(&kalmanFilter.mutex);
+    kalmanFilter.updateApogee(apogee_est);
+    chMtxUnlock(&kalmanFilter.mutex);
```

Comment:
rk4 sim apogee does take a long time, but there is no way for a different thread to gain execution while it is running, so unlocking the mutex while sim_apogee is pointless. Kalman filter also accesses the apogee estimation outside of the mutex that is supposed to be guarding it, thats a kalman filter bug though

Reply:
I'm not actually sure we're using cooperative multithreading. From what I can see of ChibiOS, there's a flag that we have to define to enable cooperative multithreading and I didn't see it anywhere.

Reply:
Actually I take that back, the fact that writing "while(true) {}" stalls the entire program means it's cooperative for unknown reasons. 

Given that that is the case, I'd say a lot of our current mutex locks are unnecessary. Should we remove those as well?

Reply:
Technically speaking the only thing our mutexs do is ensure that reads and writes actually happen. I think the goal should be to have mutexes that would work on a truly multithreaded system (though there are many places right now where that isn't true). Imo don't change mutexes as part of this pr, thats outside the scope.